### PR TITLE
Catch join call error from SDK and propagate it up to call screen

### PR DESCRIPTION
**Pull request by Gemakk**
## Purpose

* Handle exceptions thrown by the Calling SDK and display an error dialog.

## Does this introduce a breaking change?

```
[ ] Yes
[x] No
```

## Pull Request Type
What kind of change does this Pull Request introduce?

```
[x] Bugfix
[ ] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Documentation content changes
[ ] Other... Please describe:
```

### CODE REVIEW

File: AzureCalling/app/src/main/java/com/azure/samples/communication/calling/external/calling/CallingContext.java [hunk 1]
```diff
@@ -401,7 +410,11 @@ private void callWithOptions(
         final JoinCallOptions joinCallOptions = new JoinCallOptions();
         joinCallOptions.setVideoOptions(videoOptions);
         joinCallOptions.setAudioOptions(audioOptions);
-        call = agent.join(appContext, groupCallLocator, joinCallOptions);
+        try {
+            call = agent.join(appContext, groupCallLocator, joinCallOptions);
+        } catch (Exception e) {
+            throw e;
+        }
```
Comment (waynemo) [inline AzureCalling/app/src/main/java/com/azure/samples/communication/calling/external/calling/CallingContext.java:417]: If `agent.join` is throwing a RuntimeException, we should not need all the try/catch blocks.

**Review by waynemo — CHANGES_REQUESTED** [review]

File: AzureCalling/app/src/main/java/com/azure/samples/communication/calling/external/calling/CallingContext.java (hunk 1 above)
Comment (Gemakk) [reply AzureCalling/app/src/main/java/com/azure/samples/communication/calling/external/calling/CallingContext.java]: Got it!

**Review by iaulakh — DISMISSED** [review]

File: AzureCalling/app/src/main/java/com/azure/samples/communication/calling/external/calling/CallingContext.java [hunk 2]
```diff
@@ -190,10 +190,19 @@ public String getJoinId() {
...
+                        throw e;
+                    }
                 });
             } else {
-                callWithOptions(agent, audioOptions, null, callLocator);
+                try {
+                    callWithOptions(agent, audioOptions, null, callLocator);
+                } catch (Exception e) {
+                    throw e;
+                }
```
Comment (waynemo) [inline AzureCalling/app/src/main/java/com/azure/samples/communication/calling/external/calling/CallingContext.java:205]: I think these don't need to be in a try/catch either.

**Review by waynemo — CHANGES_REQUESTED** [review]

File: AzureCalling/app/src/main/java/com/azure/samples/communication/calling/external/calling/CallingContext.java (hunk 2 above)
Comment (Gemakk) [reply AzureCalling/app/src/main/java/com/azure/samples/communication/calling/external/calling/CallingContext.java]: That's right

**Review by waynemo — APPROVED** [review]

**Review by iaulakh — APPROVED** [review]

**Pull request merged**